src/htcondor_es/AffiliationManager.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Author: Christian Ariza <christian.ariza AT gmail [DOT] com>
+# pylint: disable=line-too-long
+import os
+import json
+import requests
+from datetime import datetime, timedelta
+
+
+class AffiliationManager():
+    __DEFAULT_URL = 'https://cms-cric-dev.cern.ch/api/accounts/user/query/?json'
+    __DEFAULT_DIR_PATH = 'dir.pkl'
+
+    def __init__(self,
+                 pickle_file=__DEFAULT_DIR_PATH,
+                 recreate=False,
+                 recreate_older_days=None,
+                 service_url=__DEFAULT_URL):
+        """
+        params:
+            recreate: boolean
+            recreate_older_days: int, recreate the dir if is older
+                    than that number of days.
+        """
+        self.path = pickle_file
+        self.url = service_url
+        if not recreate\
+           and recreate_older_days \
+           and os.path.isfile(self.path):
+            _min_date = datetime.now() - timedelta(days=recreate_older_days)
+            _dir_time = datetime.fromtimestamp(os.path.getmtime(self.path))
+            recreate = _dir_time < _min_date
+        self.__dir = self.loadOrCreateDirectory(recreate)
+
+    def loadOrCreateDirectory(self, recreate=False):
+        """
+        Create or load from a json file an inverted
+        index of instutions by person login. e.g.:
+
+        {
+            'valya':{u'country': u'US',
+                     u'institute': u'Cornell University'},
+            'belforte': {u'country': u'IT',
+                         u'institute': u'Universita e INFN Trieste'}
+            ...
+        }
+        """
+        _tmp_dir = None
+        if not os.path.isfile(self.path) or recreate:
+            with open(self.path, 'wb') as _dir_file:
+                response = requests.get(self.url, verify=False)
```

Comment:
I supposed this call can take a while, so it might be better to create the dictionary in a separate process outside of the spider. Maybe in a cron job once a day?

Reply:
If the file exists and doesn't need to recreate it only takes around 25ms. If need to make the request, it can take a while(at least 3 seconds in the tests I did), but it only happens once, before it starts to process the queues. (The alert signal can be a problem, but I moved its definition after the creation of the object).

Reply:
I would still recommend to separate this part and add some protection for when the request fails. Something like: a separate process creates the lookup table from the request once a day. If the request fails, it falls back to the previous table. The spider just loads the table. If the loading of the table fails or it is empty, a logging warning is issued (maybe also an email warning?) and the information is not filled in the documents.

Reply:
Just committed the changes. I tested it using both _testDocConversion_ and sending documents to es-cms (either when the file exists and when it doesn't). 
![image](https://user-images.githubusercontent.com/4405564/57708731-202de380-766a-11e9-94db-e5bbaae90ae3.png)
I also created "cms-htcondor-es/cronAggregation.sh" script to be run with cron.